### ISSUE TITLE
default.nix vs shell.nix

### ISSUE THREAD

**Issue by tomberek** [NixOS/nix.dev#135]
A common problem is overloading `nix-shell` with two responsibilities, to establish a dev environment and to build necessary software. A first step I've done taken is to help create a users split these concepts in two with separate `default.nix` and `shell.nix` to keep things organized. The new Nix UI with `nix develop`/`nix shell`/etc. and the standard organization flakes provides can help here.

**Comment by fricklerhandwerk** [comment]
For a more recent state of discussion on this topic: https://github.com/NixOS/nix/pull/4702#issuecomment-2233787312

tl;dr I don't think Nix should know about any of this, and neither users nor documentation maintainers should have to put up with Nix trying to be too clever. Unless we want to keep allowing cyclic dependencies, something like `nix shell` and `nix develop` should be Nixpkgs facilities. See the [Nixpkgs CLI working group idea](https://discourse.nixos.org/t/nixpkgs-cli-working-group-member-search/30517) for more of that.

Closing the issue since, in my opinion, it's a problem that doesn't have to exist and can still be avoided to be set in stone by letting go of most of the new Nix CLI and building more sustainable solutions in Nixpkgs.
